src/MySqlConnector/Logging/ConsoleLoggerProvider.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Text;
+
+namespace MySqlConnector.Logging
+{
+	public class ConsoleLoggerProvider : IMySqlConnectorLoggerProvider
+	{
+		public ConsoleLoggerProvider(MySqlConnectorLogLevel minimumLevel = MySqlConnectorLogLevel.Info, bool isColored = true)
+		{
+			if (minimumLevel < MySqlConnectorLogLevel.Trace || minimumLevel > MySqlConnectorLogLevel.Fatal)
+				throw new ArgumentOutOfRangeException(nameof(minimumLevel), "minimumLevel must be between Trace and Fatal");
+
+			m_minimumLevel = minimumLevel;
+			m_isColored = isColored;
+		}
+
+		public IMySqlConnectorLogger CreateLogger(string name) => new ConsoleLogger(this, name);
+
+		private sealed class ConsoleLogger : IMySqlConnectorLogger
+		{
+			public ConsoleLogger(ConsoleLoggerProvider provider, string name)
+			{
+				m_provider = provider;
+				m_name = name;
+			}
+
+			public bool IsEnabled(MySqlConnectorLogLevel level) => level >= m_provider.m_minimumLevel && level <= MySqlConnectorLogLevel.Fatal;
+
+			public void Log(MySqlConnectorLogLevel level, string message, object[] args = null, Exception exception = null)
+			{
+				if (!IsEnabled(level))
+					return;
+
+				var sb = new StringBuilder();
+				sb.Append(s_levels[(int) level]);
+				sb.Append('\t');
```

Comment:
Why not just a space? Tabs inconsistently waste horizontal space on a console.

Reply:
This is not a "production quality" logger; it's a sample implementation that can be used for testing. To get console output formatted the way you would like, hook `IMySqlConnectorLogger` up to log4net and use a `ConsoleAppender` (sample code/NuGet package for that hopefully coming soon).

> Why not just a space?

I prefer tabs in my console output; keeps the columns lined up.